File: selection-sort.java


class Solution
{
	int  select(int arr[], int i)
	{
        // code here such that selectionSort() sorts arr[]
        // System.out.println(i + arr.toString());
         int min_index = i;
        
        for(int j = i+1; j < arr.length; j++){
            
            if(arr[min_index] > arr[j]){
                min_index = j;
            }
        }
        
           int temp = arr[i];
           arr[i] = arr[min_index];
           arr[min_index] = temp;
           return 0;
	}
	
	void selectionSort(int arr[], int n)
	{
	    
	    //code here
	     for( int i = 0; i < n-1; i++)
	    {
        select(arr,i);
        }
	}
}

File: Selection Sort - GFG/selection-sort.py
#User function Template for python3

class Solution: 
    def select(self, arr, i):
        # code here 
        
        size = len(arr)
        min_idx = i
        for j in range(i + 1, size):
            if arr[min_idx] > arr[j]:
                min_idx = j
        arr[i], arr[min_idx] = arr[min_idx], arr[i]

    
    def selectionSort(self, arr,n):
        #code here
        size = len(arr)
        for i in range(size):
            self.select(arr, i)

       

#{ 
 # Driver Code Starts
#Initial Template for Python 3

if __name__ == '__main__': 
    t = int(input())
    for _ in range(t):
        n = int(input())
        arr = list(map(int, input().strip().split()))
        Solution().selectionSort(arr, n)
        for i in range(n):
            print(arr[i],end=" ")
        print()
# } Driver Code Ends